api/src/middlewares/services/ses.ts

```diff
@@ -0,0 +1,121 @@
+import logger, { Logger } from "pino";
+import config from "config";
+import {
+  GetTemplateCommand,
+  SendRawEmailCommand,
+  SESClient,
+  Template,
+} from "@aws-sdk/client-ses";
+import { ERRORS } from "../../errors";
+import { createMimeMessage } from "mimetext";
+import {
+  AffirmationTemplateData,
+  CNITemplateData,
+} from "../../handlers/helpers/getTemplateDataFromInputs";
+import { ses } from "../../SESClient";
+
+export class SESService {
+  templates: {
+    cni?: string;
+    affirmation?: string;
+  } = {};
+  logger: Logger;
+  ses: SESClient;
+
+  constructor() {
+    this.templates.affirmation = config.get("affirmationTemplate");
+    this.templates.cni = config.get("cniTemplate");
+    this.logger = logger().child({ service: "SES" });
+    this.ses = ses;
+  }
+
+  async getTemplate(type: "cni" | "affirmation") {
+    try {
+      if (!this.templates[type]) {
+        this.logger.error(ERRORS.ses.NO_TEMPLATE);
+        return;
+      }
+      const templateData = await this.ses.send(
+        new GetTemplateCommand({
+          TemplateName: this.templates[type],
+        })
+      );
+
+      if (!templateData?.Template) {
+        this.logger.error(ERRORS.ses.TEMPLATE_NOT_FOUND);
+        return;
+      }
+      return templateData.Template;
+    } catch (err) {
+      console.log("handling the error");
+      this.handleSESError(err as Error);
+    }
+    return;
+  }
+
+  buildEmail(
+    template: Template,
+    fields: AffirmationTemplateData | CNITemplateData,
+    uploads: object
+  ) {
+    const message = createMimeMessage();
+    message.setSender({
+      name: "Getting Married Abroad Service",
+      addr: config.get("senderEmail"),
+    });
+    message.setRecipient(config.get("submissionAddress"));
+    try {
+      message.setSubject(this.interpolateVars(template.SubjectPart, fields));
+      for (const [field, file] of Object.entries(uploads)) {
+        message.setAttachment(field, "application/pdf", file.toString());
+      }
+      message.setMessage(
+        "text/html",
+        this.interpolateVars(template.HtmlPart, fields)
+      );
+    } catch (err) {
+      return {
+        errors: err as Error,
+      };
+    }
+    return message.asRaw();
+  }
+
+  async sendEmail(message: string) {
+    try {
+      return await this.ses.send(
+        new SendRawEmailCommand({
+          RawMessage: {
+            Data: Buffer.from(message),
+          },
+        })
+      );
+    } catch (err) {
+      this.handleSESError(err as Error);
+    }
+    return;
+  }
+
+  interpolateVars(string: string = "", vars: object) {
```

Comment:
I'd either install handlebars to render the template, or use Nodemailer which should handle this + allow you to attach easily